src/DigitalSubAuth.js

```diff
@@ -0,0 +1,98 @@
+const AWS = require('aws-sdk')
+const moment = require('moment')
+
+AWS.config.update({
+    region: "eu-west-1"
+})
+
+const stages = ['CODE', 'PROD', 'DEV']
+let stage = stages.find((stage) => {
+    return stage === process.env.Stage
+})
+const dynamo = new AWS.DynamoDB({apiVersion: '2012-08-10'})
+const dynamoTableName = `cas-auth-${stage}`
+
+async function setInDynamo(appId, deviceId, expiry, TTL) {
+    const params = {
+        TableName: dynamoTableName,
+        Item: {
+            "appId": {"S": appId},
+            "deviceId": {"S": deviceId},
+            "expiryDate": {"S": expiry},
+            "ttlTimestamp": {"N": TTL}
+        }
+    }
+    await dynamo.putItem(params).promise()
+}
+
+function isEmpty(obj) {
+    for (var key in obj) {
+        if (obj.hasOwnProperty(key))
+            return false
+    }
+    return true
+}
+
+async function getExpiryFromDynamo(appId, deviceId) {
+    const params = {
+        TableName: dynamoTableName,
+        AttributesToGet: ["expiryDate"],
+        Key: {
+            "appId": {"S": appId},
+            "deviceId": {"S": deviceId}
+        }
+    }
+    let dynamoResponse = await dynamo.getItem(params).promise()
+    if (isEmpty(dynamoResponse)) {
+        return null //todo what's a nice way of doing something like options in js ?
```

Comment:
You could do:
`Object.keys(dynamoResponse).length > 0` 
or
`if (dynamoResponse.Item) { return dynamoResponse.Item.expiryDate.S } else { return null }`
or 
```
const item = dynamoResponse['Item'] || {}
const dateObj = item['expiryDate'] || {}
return dateObj.S || null;
```